### DISCUSSION TITLE
Adding support for DIY lorawan gateways

### DISCUSSION THREAD

**Discussion by T00ManySecrets** [markqvist/Reticulum#491] [Ideas]
I build a few gateways in my time like this
https://www.thethingsnetwork.org/docs/gateways/start/build/
Not knowing to much about the backend side of Reticulum id like to know is there any benifit of supporting these kind of devices as a transport layer?
This [https://lora-alliance.org/wp-content/uploads/2019/09/iC880A_Datasheet_V1_1.pdf](Concentrator) works over SPI and can run perfectly with some low power boards. I have one running on a Orange Pi Zero for more than 5 years now.

**Comment by faragher** [comment]
To your larger question, I don't know if there's a benefit. 

My understanding is LoRaWAN is an IoT/M2M system which uses LoRa but is mostly defined by the service providers. If there's an uplink from a LoRaWAN provider to a larger Reticulum network, then it might be useful if all else is equal. However, I seem to recall, as a shared service, there's significant regulation as to throughput and other ways they attempt to make the service available to everyone, not just a few noisy people. IIRC, this includes registering with their systems.

Now, I could be wrong about any of those things, and I could be conflating a protocol with a service. You would know better than I would. 

If, in either case, there's an MDU of at least 500 and you'll be able to reliably send and receive multiple packets to an arbitrary data size, then it's likely technically possible, and if it's useful depends on the user's needs.

If you're interested, I'd suggest looking at making a new Interface for Reticulum, like the TCP or RNode interfaces, and working with the LoRaWAN devices directly. If they support serial data in or AX.25 style commands, it may be incredibly straightforward.

However, once again, I'm not familiar enough with LoRaWAN to provide a GO/NO-GO on a technical level, or discuss if it's beneficial in general. However, if you think it's at least an interesting idea, looking into feasibility or even building a prototype Interface would be much appreciated. If it's suboptimal but successful, someone will be interested, and if it's a generally bad idea, someone will ask again eventually, and it's nice to show them the last attempt if they wish to try to continue.
